graph/history.go

```diff
@@ -8,6 +9,78 @@ import (
 	"github.com/docker/docker/utils"
 )
 
+// History returns the list of all images used to create this image.
+func (graph *Graph) History(img *image.Image) ([]*image.Image, error) {
```

Comment:
The `*image.Image` is not altered after creation. Obviously there is nothing to protect it, but that pattern is used throughout the entire package. Would be better to protect with an interface than just change a large struct to pass by value.

The `History` is not actually called anywhere, I only kept it because it appears to have a clear purpose.